js/postDetails.js

```diff
@@ -0,0 +1,61 @@
+// postDetails.js
+import { fetcher } from "./fetcher.js";
+import { BASE_API_URL } from "./common/constants.js";
+
+document.addEventListener("DOMContentLoaded", () => {
+  // Get post ID from the URL
+  const postId = new URLSearchParams(window.location.search).get("id");
+
+  // Fetch post details
+  const apiUrl = `${BASE_API_URL}/social/posts/${postId}`;
+
+  fetcher(apiUrl, { method: "GET" }, true)
```

Comment:
It's better to be consistent in the way you make API calls. Here you are using then(), but in other files you are using async/await. Stick to one way, the one that makes most sense to you.